bot/exts/fun/quack.py

```diff
@@ -0,0 +1,62 @@
+from typing import Optional
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+API_URL = 'https://quackstack.pythondiscord.com'
+
+
+class Quack(commands.Cog):
+    """Cog for the .quack command."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    @commands.command()
+    async def quack(self, ctx: commands.Context, ducktype: Optional[str] = "duck", seed: Optional[str] = None) -> None:
+        """
+        Uses the Quackstack API to generate a random duck.
+
+        If a seed is provided, a duck is generated based on the given seed.
+        If a duck type is provided, either a duck or manduck will be generated.
+        """
+        ducktype = ducktype.lower()
+        if ducktype not in ["duck", "manduck"]:
+            raise commands.BadArgument(
+                "The duck type can only be `duck` or `manduck`."
+            )
+
+        quackstack_url = f"{API_URL}/{ducktype}"
+        params = {}
+        if seed is not None:
+            try:
+                seed = int(seed)
+            except ValueError:
+                newseed = ""
+                for e in seed:
+                    newseed += str(ord(e))
+                seed = int(newseed)
+            params["seed"] = seed
+
+        async with self.bot.http_session.get(quackstack_url, params=params) as r:
+            data = await r.json()
+            file = data.get("file")
+            image_url = f"{API_URL}{file}"
```

Comment:
This doesn't handle down-time, even though it should be little and we should know when that is. Unneccessary to blow up the console about it.